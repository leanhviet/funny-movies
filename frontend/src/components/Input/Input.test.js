// Libs
import React from 'react'
import { shallow } from 'enzyme'

// Components
import { InputComponent } from './index'

const props = {
  id: '123',
  type: 'text',
  className: 'input-fmt',
  name: 'input-txt',
  defaultValue: 'test',
  placeholder: 'Enter your email',
  disabled: false,
  size: 'lg',
  minlength: 8,
  required: false
}

describe('Input component', () => {
  const wrapper = shallow(<InputComponent />)
  const wrapperWithProps = shallow(<InputComponent {...props} />)

  it('should be defined Input component', () => {
    expect(wrapper).toBeDefined()
  })

  // Testing Snapshot
  it('should be render snapshot Input component', () => {
    expect(wrapper).toMatchSnapshot()
    expect(wrapperWithProps).toMatchSnapshot()
  })

  // // Testing default props
  it('should be render defaultProps Input component', () => {
    expect(InputComponent.defaultProps.className).toEqual('')
    expect(InputComponent.defaultProps.type).toEqual('text')
    expect(InputComponent.defaultProps.defaultValue).toEqual('')
    expect(InputComponent.defaultProps.placeholder).toEqual('')
  })

  // // Testing mock props
  it('should be render mockProps Input component', () => {
    expect(wrapperWithProps.find('FormControl').props().className).toContain(
      props.className
    )
    expect(wrapperWithProps.find('FormControl').props().placeholder).toEqual(
      props.placeholder
    )
    expect(wrapperWithProps.find('FormControl').props().type).toEqual(
      props.type
    )
    expect(wrapperWithProps.find('FormControl').props().defaultValue).toEqual(
      props.defaultValue
    )
  })
})
