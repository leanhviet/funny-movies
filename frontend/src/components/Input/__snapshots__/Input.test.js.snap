// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should be render snapshot Input component 1`] = `
<FormControl
  className="form-control "
  defaultValue=""
  name=""
  onBlur={[Function]}
  onChange={[Function]}
  placeholder=""
  required={false}
  type="text"
/>
`;

exports[`Input component should be render snapshot Input component 2`] = `
<FormControl
  className="form-control input-fmt"
  defaultValue="test"
  disabled={false}
  id="123"
  minlength={8}
  name="input-txt"
  onBlur={[Function]}
  onChange={[Function]}
  placeholder="Enter your email"
  required={false}
  size="lg"
  type="text"
/>
`;
